core-container/entry.sh

```diff
@@ -16,88 +16,487 @@
 
 set -e
 
-echo "< Starting Kubestellar container >-------------------------"
+KUBESTELLAR_SERVICE="kubestellar"
 
-# Try to guess the route/ingress
-if [ -z "$EXTERNAL_HOSTNAME" ]; then
-    if kubectl get route kubestellar-route &> /dev/null; then
-        EXTERNAL_HOSTNAME=$(kubectl get route kubestellar-route -o yaml -o jsonpath={.spec.host} 2> /dev/null)
+
+function echoerr() {
+   echo "ERROR: $1" >&2
+}
+
+
+function ensure_inv() {
+    while IFS=':' read -ra items; do
```

Comment:
Now that we avoid workspace hierarchy this can be much simpler.

Reply:
I'm not sure I understand